extensions/parquet/table/src/main/java/io/deephaven/parquet/table/layout/ParquetKeyValuePartitionedLayout.java

```diff
@@ -88,8 +89,16 @@ public ParquetKeyValuePartitionedLayout(
     public final void findKeys(@NotNull final Consumer<ParquetTableLocationKey> locationKeyObserver) {
         final SeekableChannelsProvider provider = SeekableChannelsProviderLoader.getInstance().fromServiceLoader(
                 tableRootDirectory, readInstructions.getSpecialInstructions());
+        final boolean isFileURI = FILE_URI_SCHEME.equals(tableRootDirectory.getScheme());
         try (final Stream<URI> uriStream = provider.walk(tableRootDirectory)) {
-            final Stream<URI> filteredStream = uriStream.filter(ParquetUtils::isVisibleParquetURI);
+            final Stream<URI> filteredStream = uriStream.filter(uri -> {
+                if (isFileURI) {
```

Comment:
I might suggest pulling the bifurcation out of the inner loop; either, creating a `Predicate<URI>`, or having two different call sites to `.filter`.